src/formatDate.js

```diff
@@ -50,7 +50,34 @@
  */
 
 function formatDate(date, fromFormat, toFormat) {
-  // write code here
+  const inputDate = date.split(fromFormat[3]);
+  const number = inputDate[fromFormat.indexOf('DD')];
+  const month = inputDate[fromFormat.indexOf('MM')];
+  const year = inputDate[fromFormat.indexOf('YYYY')]
+  || inputDate[fromFormat.indexOf('YY')];
+
+  const copyToFormat = [...toFormat];
+  const dateBreakPoint = 30;
+  const currentCentury = 20;
+  const lastCentury = 19;
+
+  copyToFormat[copyToFormat.indexOf('DD')] = number;
+  copyToFormat[copyToFormat.indexOf('MM')] = month;
+
+  copyToFormat[copyToFormat.indexOf('YYYY')] = (year.length === 4)
+    ? year
+    : (year < dateBreakPoint)
+      ? currentCentury + year
+      : lastCentury + year;
+
+  copyToFormat[copyToFormat.indexOf('YY')] = (year.length === 2)
+    ? year
+    : year.slice(2);
+
+  const resultData = copyToFormat[0] + copyToFormat[3] + copyToFormat[1]
+    + copyToFormat[3] + copyToFormat[2];
```

Comment:
😵‍💫
it's hard to understand what you are doing here, would be better if you will do this with 'copyToFormat' and in the end just 'join' him